### PR TITLE
refactoring ramps to their own component

### PR DESCRIPTION
**Pull request by ongspxm**
```
Currently, the entire ramp template resides in `v_summary`. In light of
the "zoomin" tab view, we would have to reuse ramp chart view to
display the commits in the "zoomin" tab.

Let's refactor to move the ramp into its own component, so as to
support reuse in the "zoomin" tab view.
```

### CODE REVIEW

File: frontend/src/index.pug [hunk 1]
```diff
@@ -167,6 +142,9 @@ html
                   v-bind:style="{ width: width+'%' }"
                 )
 
+    vuetemplate#v_ramp
+      include ramp.pug
```
Comment (chelseyong) [inline frontend/src/index.pug]: Why the change from `jade` to `pug`?

**Comment by ongspxm** [comment]
https://github.com/pugjs/pug/issues/2184 haha, it still works, but the
compiler will complain that the name should be .pug, so change lorh

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -70,7 +70,7 @@ Eslint and its accompaning modules can be installed through NPM, so do ensure th
 
 1. Execute the following command on the OS terminal inside the project directory. <br/>
 Usage: `gradlew run -Dargs="([--config CONFIG_FOLDER] | [--repos REPO_PATH_OR_URL...]) [--view [REPORT_FOLDER]] [--output OUTPUT_DIRECTORY] [--since DD/MM/YYYY] [--until DD/MM/YYYY] [--formats FORMAT...] [--ignore-standalone-config]"` <br/>
-Named Arguments: 
+Named Arguments:
```
Comment (eugenepeh) [inline docs/DeveloperGuide.md]: unnecessary change

Comment (ongspxm) [reply docs/DeveloperGuide.md]: fixed

**Comment by ongspxm** [comment]
@yamidark @eugenepeh merge pls
